src/Microsoft.Azure.SignalR.Common/Auth/IAadAuthOptions.cs

```diff
@@ -0,0 +1,9 @@
+﻿using System.Threading.Tasks;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal interface IAadAuthOptions
```

Comment:
I think `Options` in aspnetcore has some special meaning, it usually means the Options obtained from Configuration, how about changing it to some `IGenerator`?